services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/gdpr/KortistoGdprServiceITest.kt

```diff
@@ -240,8 +241,8 @@ class KortistoGdprServiceITest(
         fun `returns true when there are only draft applications`() {
             val hanke = hankeFactory.saveWithAlue(USERNAME)
             val kayttaja = hankeKayttajaService.getKayttajaByUserId(hanke.id, USERNAME)!!
```

Comment:
Also not really part of this PR, but as we discussed about service functions using entities as parameters, you mentioned that we should not return an entity from a service method. Maybe at some point we will refactor these `HankeKayttajaService` functions that currently return entities?

Reply:
I think that one needs to return an entity since it's needed as a reference in other tables. The alternative would be to copy-paste the code to the other services that need it, which wouldn't be ideal.

Maybe it would be useful to set the `@Transactional` propagation to `MANDATORY` here?

The uses in tests seem to all be for getting the founder of a hanke, so there could be a custom test method for that.

But I'm not sure how to test `getKayttajaByUserId` itself since using `@Transactional` in tests has big caveats.

Reply:
Maybe we don't change anything for this now. Just have to remember that in some cases returning Entity is a valid option. But usually not and it needs some consideration whether to use it or not.